api/src/main/java/org/apache/cloudstack/api/command/user/template/RegisterTemplateCmd.java

```diff
@@ -275,7 +280,8 @@ public boolean isDirectDownload() {
     }
 
     public Boolean isDeployAsIs() {
-        return hypervisor != null && hypervisor.equalsIgnoreCase(Hypervisor.HypervisorType.VMware.toString());
+        return hypervisor != null && hypervisor.equalsIgnoreCase(Hypervisor.HypervisorType.VMware.toString())
+                && deployAsIs != null && deployAsIs;
```

Comment:
```suggestion
        return Hypervisor.HypervisorType.VMware.toString().equalsIgnoreCase(hypervisor) &&
                Boolean.TRUE.equals(deployAsIs);
```
Somewhat easier to read